components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -579,7 +589,7 @@ interface JsonRpcService {
   GetNetworkUrl() => (string network_url);
 
   // Obtains the native balance (e.g. ETH for Ethereum) for the address
-  GetBalance(string address) => (string balance, ProviderError error, string error_message);
+  GetBalance(string address, CoinType coin) => (string balance, ProviderError error, string error_message);
```

Comment:
Would `chainId` will be a better choice here, like @bbondy introduced in this PR: https://github.com/brave/brave-core/pull/11791? It looks like a good opportunity to start decoupling service methods from the currently selected network.

Reply:
no, that won't work when we add Solona later. it is better to follow `coin_type` of bip44

Reply:
actually let's move keep this API as it is now and add `chainId` later and change the name to `GetETHBalance` because this API is dedicated to ETH
Add a new one `GetFILBalance` for Filecoin and we will have `GetSOLBalance` later

Reply:
FYI I am adding a new GetSolanaBalance in the interface https://github.com/brave/brave-core/pull/11836/files#diff-9f2cf11d57c8efee5461b92cdb38fa327b70c25fd8d009f7692049e8e88944a8R636.
For solana's case, we only have a few to be added and won't pollute this interface by adding that few of methods, and there are a separate set of possible error codes too, which will be conflict with current ProviderError, so I think it makes more sense to just have a separate `GetSolanaBalance` method.

Reply:
created issue for this https://github.com/brave/brave-browser/issues/20496

Reply:
Sounds good to me.